feat: reviews

art1722 opened the pull request:
- add a review page

- add review stats and review records for a provider

<img width="1874" height="1083" alt="2025-11-13 20_42_22-ProviderProfilePage tsx - se_project_ban_saat - Visual Studio Code" src="https://github.com/user-attachments/assets/f62d47b1-a6ac-4505-9d6e-98240e5cd4f8" />

<img width="1920" height="1086" alt="2025-11-13 21_19_09-Chat tsx - se_project_ban_saat - Visual Studio Code" src="https://github.com/user-attachments/assets/cc340aa2-f7e2-42db-9340-a52d5a90686a" />

art1722 committed:
feat: add review page, add reviews in the provider's profile

BadLuckZ commented on src/api/review.ts:
```diff
@@ -0,0 +1,27 @@
+type ApiResult<T> = {
+  success: boolean;
+  data: T;
+};
```
src/config/api.ts มี ResponseInterface นะ เหมือนกันเป๊ะเลย แล้ว file อื่นๆ ใน directory นี้ก็ใช้งี้ แต่แก้ไม่แก้ไม่ซีเรียสคับ

art1722 committed:
feat: add review button on the chat page after completing the payment

art1722 committed:
fix: fix bug in redirecting page after submitting a review

art1722 commented:
I've added a feature. When a payment succeeds, the chat page now shows a "Write a Review" button. I’ve also updated the code to ResponseInterface.

<img width="1916" height="1085" alt="2025-11-14 00_57_38-http___localhost_5000_api_payments_dev_force-succeeded_68efb4fd37a782fead1e9c18 " src="https://github.com/user-attachments/assets/642eb323-6f92-4cd8-9013-893ee73df90f" />

art1722 committed:
fix: user can create only one review per post & avatarProfile bug fixes

art1722 committed:
ui: resizable summarized review plot

art1722 committed:
ui: add loading page while checking parameters

art1722 committed:
refactor: fix spacing

Friend186 reviewed (approved):
great

dannifyz committed:
Merge branch 'developer' of https://github.com/PHP-BEST/baan-saat-frontend into feat/reviews
